src/system/process.py

```diff
@@ -58,15 +58,33 @@ def terminate(self) -> int:
         logging.info(f"Process killed with exit code {self.process.returncode}")
 
         if self.stdout:
-            self.__stdout_data__ = open(self.stdout.name, 'r').read()
+            self.stdout.seek(0)
+            self.__stdout_data__ = self.stdout.read()
+            self.stdout.flush()
```

Comment:
I can understand the flush but why do we need seek(0) here?

Reply:
From my understanding, after we `write()` to the tempfile, basically the file handle would be at the end of the file. Then when we `read` from it, we would get nothing from the end. Therefore moving the file handle to the beginning position would enable us to read the complete file. 
Check this example as well. https://docs.python.org/3/library/tempfile.html#examples